### DISCUSSION TITLE
How to open file in a `vsplit` when using `current_window`?

### DISCUSSION THREAD

**Discussion by enheit** [nvim-tree/nvim-tree.lua#2818] [Q&A]
This is my current nvim-tree configuration

```lua
return {
  "nvim-tree/nvim-tree.lua",
  version = "*",
  lazy = false,
  dependencies = {
    "nvim-tree/nvim-web-devicons",
  },
  config = function()
    local nvimTree = require("nvim-tree")
    local api = require("nvim-tree.api")

    nvimTree.setup({
      disable_netrw = true,
      actions = {
        open_file = {
          quit_on_open = true,
        }
      },
      renderer = {
        icons = {
          git_placement = "after",
          glyphs = {
            git = {
              staged = "S",
              unstaged = "M",
              untracked = "U",
              deleted = "D"
            },
            folder = {
              arrow_closed = "",
              arrow_open = ""
            }
          }
        }
      },
      filters = {
        dotfiles = true
      }
    })

    vim.keymap.set("n", "<leader>e", function() api.tree.toggle({ current_window = true, find_file = true }) end,
      { noremap = true })
  end,
}
```

I open my nvim-tree in the current window

```lua
# cut
    vim.keymap.set("n", "<leader>e", function() api.tree.toggle({ current_window = true, find_file = true }) end,
      { noremap = true })
# cut
```

Whenever I try to open a file in a vertical split (or horizontal) by pressing `<C-x>` or `<C-v>`, it just opens the file in a full-size window, not in a split.

I noticed that if I set `quit_on_open` to `false`, it opens the file in a vertical split, but it opens next to the file tree.

Desired behavior:

After selecting a file and pressing `<C-x>` or `<C-v>`, it should close the full-size tree and open the selected file in a vertical split next to the currently opened file.

Any ideas why it doesn't work?

**Comment by alex-courtis** [comment]
I'm not able to replicate with `quit_on_open` `true` or `false`.

Please raise a [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml) with your config and full reproduction steps.

One thing of note: the split will not occur when the file is already open in an existing window, it will just switch to that window.

Splits:
![20240707_104932](https://github.com/nvim-tree/nvim-tree.lua/assets/1505378/007b5687-7b55-4354-8c17-dfb8435f4810)
`<c-x>`
![20240707_104941](https://github.com/nvim-tree/nvim-tree.lua/assets/1505378/03bc5f06-dfe4-4b31-954c-61cf8feded99)

Doesn't split:
![20240707_104908](https://github.com/nvim-tree/nvim-tree.lua/assets/1505378/daa78144-525e-448b-8e06-ecc390b3cb00)
`<c-x>`
![20240707_104958](https://github.com/nvim-tree/nvim-tree.lua/assets/1505378/60c5568c-f51d-4520-8e0d-848adf57454b)
